tests/post/features/ingress.feature

```diff
@@ -19,12 +19,32 @@ Feature: Ingress
         When we perform an HTTP request on port 80 on a control-plane IP
         Then the server should not respond
 
+    Scenario: Failover of Control Plane Ingress VIP using MetalLB
+        Given the Kubernetes API is available
+        And we are on a multi node cluster
+        And MetalLB is enabled
+        When we stop the node hosting the Control Plane Ingress VIP
+        Then the node hosting the Control Plane Ingress VIP changed
+        And we are able to login to Dex as 'admin@metalk8s.invalid' using password 'password'
+
     Scenario: Change Control Plane Ingress IP to node-1 IP
         Given the Kubernetes API is available
         And we are on a multi node cluster
         And pods with label 'app.kubernetes.io/name=ingress-nginx' are 'Ready'
-        When we update control plane ingress IP to node 'node-1' IP
+        When we disable MetalLB and set control plane ingress IP to node 'node-1' IP
         And we wait for the rollout of 'daemonset/ingress-nginx-control-plane-controller' in namespace 'metalk8s-ingress' to complete
         And we wait for the rollout of 'deploy/dex' in namespace 'metalk8s-auth' to complete
         Then the control plane ingress IP is equal to node 'node-1' IP
         And we are able to login to Dex as 'admin@metalk8s.invalid' using password 'password'
+
+    Scenario: Change Control Plane Ingress IP to a new VIP
+        Given the Kubernetes API is available
+        And a VIP for Control Plane Ingress is available
+        And pods with label 'app.kubernetes.io/name=ingress-nginx' are 'Ready'
+        When we enable MetalLB and set control plane ingress IP to '{new_cp_ingress_vip}'
```

Comment:
Same here IMO, are we testing the **VIP change while MetalLB is enabled**, or **enabling MetalLB**? We shouldn't do both in the same scenario.

Reply:
It does not make sense to just "enable MetalLB" you **must** provide a VIP when you enable MetalLB

Reply:
I see your point we should have a test that "Given MetalLB is disabled then we switch to MetalLB using this IP" and a test "Given MetalLB is enabled then we change the VIP"
But to me it does not bring any value since all of this is already tested as part of other tests (including teardown of tests) so it will just make the test last longer for ... no reason

Reply:
OK, agreed, the "VIP change while MetalLB is enabled" is tested in the teardown of this test (that's only true because we start the test suite from a state where MetalLB is enabled, but let's admit that's enough for now).

I think this scenario should be slightly updated to reflect the intent better:

```diff
-    Scenario: Change Control Plane Ingress IP to a new VIP
+    Scenario: Enable MetalLB
         Given the Kubernetes API is available
+        And MetalLB is disabled
         And a VIP for Control Plane Ingress is available
         And pods with label 'app.kubernetes.io/name=ingress-nginx' are 'Ready'
         When we enable MetalLB and set control plane ingress IP to '{new_cp_ingress_vip}'
         And we wait for the rollout of 'deploy/metallb-controller' in namespace 'metalk8s-loadbalancing' to complete
         And we wait for the rollout of 'daemonset/metallb-speaker' in namespace 'metalk8s-loadbalancing' to complete
         And we wait for the rollout of 'deploy/ingress-nginx-control-plane-controller' in namespace 'metalk8s-ingress' to complete
         And we wait for the rollout of 'deploy/dex' in namespace 'metalk8s-auth' to complete
         Then the control plane ingress IP is equal to '{new_cp_ingress_vip}'
         And we are able to login to Dex as 'admin@metalk8s.invalid' using password 'password'
```

Reply:
I can do this sure, the test will then really be "Enable MetalLB" I totally agree but it will last longer (which is not really an issue I think) and also if we do what you suggest we will have those tests covered in ingress

If MetalLB enabled initially:
1- Switch to non-MetalLB + Change CP Ingress IP + (teardown) switch to MetalLB
2- Switch to non-MetalLB + Switch to MetalLB + (teardown) change MetalLB IP
If MetalLB disabled initially:
1- Change CP Ingress IP + (teardown) change again CP Ingress IP
2- Switch to MetalLB + (teardown) switch back to non MetalLB

Where with my current approach:
 
If MetalLB enabled initially:
1- Switch to non-MetalLB + Change CP Ingress IP + (teardown) switch to MetalLB
2- Change MetalLB IP + (teardown) change MetalLB IP
If MetalLB disabled initially:
1- Change CP Ingress IP + (teardown) change again CP Ingress IP
2- Switch to MetalLB + (teardown) switch back to non MetalLB

So ... the list of what is tested is the same at the end but I agree you approach is likely cleaner, I will change it :+1: